src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/JsonCollectionConverter.cs

```diff
@@ -141,6 +142,8 @@ protected static JsonConverter<TElement> GetElementConverter(ref WriteStack stat
                     {
                         ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(TypeToConvert);
                     }
+
+                    state.Current.JsonPropertyInfo = elementTypeInfo.PropertyInfoForTypeInfo;
```

Comment:
Why's this being repeated here?

Reply:
Different paths -- first one is fast path deserialization, second is async deserialization.